src/JetMETAnalyzer.cc

```diff
@@ -0,0 +1,517 @@
+  std::vector<int>& refpdgid_algorithmicDef = tree["refpdgid_algorithmicDef"].write<std::vector<int>>();
+  std::vector<int>& refpdgid_physicsDef = tree["refpdgid_physicsDef"].write<std::vector<int>>();
+  std::vector<int>& refpdgid = tree["refpdgid"].write<std::vector<int>>();
+  std::vector<float>& refdrjt = tree["refdrjt"].write<std::vector<float>>();
+  std::vector<float>& refe = tree["refe"].write<std::vector<float>>();
+  std::vector<float>& refpt = tree["refpt"].write<std::vector<float>>();
+  std::vector<float>& refeta = tree["refeta"].write<std::vector<float>>();
+  std::vector<float>& refphi = tree["refphi"].write<std::vector<float>>();
+  std::vector<float>& refm = tree["refm"].write<std::vector<float>>();
+  std::vector<float>& refy = tree["refy"].write<std::vector<float>>();
+  std::vector<float>& refarea = tree["refarea"].write<std::vector<float>>();
+  std::vector<float>& jte = tree["jte"].write<std::vector<float>>();
+  std::vector<float>& jtpt = tree["jtpt"].write<std::vector<float>>();
+  std::vector<float>& jteta = tree["jteta"].write<std::vector<float>>();
+  std::vector<float>& jtphi = tree["jtphi"].write<std::vector<float>>();
+  std::vector<float>& jtm = tree["jtm"].write<std::vector<float>>();
+  std::vector<float>& jty = tree["jty"].write<std::vector<float>>();
+  std::vector<float>& jtarea = tree["jtarea"].write<std::vector<float>>();
+  std::vector<float>& jtjec = tree["jtjec"].write<std::vector<float>>();
+
+  std::vector<float>& beta = tree["beta"].write<std::vector<float>>();
+  std::vector<float>& betaStar = tree["betaStar"].write<std::vector<float>>();
+  std::vector<float>& betaClassic = tree["betaClassic"].write<std::vector<float>>();
+  std::vector<float>& betaStarClassic = tree["betaStarClassic"].write<std::vector<float>>();
+  std::vector<float>& dZ = tree["dZ"].write<std::vector<float>>();
+  std::vector<float>& DRweighted = tree["DRweighted"].write<std::vector<float>>();
+  std::vector<float>& fRing0 = tree["fRing0"].write<std::vector<float>>();
+  std::vector<float>& fRing1 = tree["fRing1"].write<std::vector<float>>();
+  std::vector<float>& fRing2 = tree["fRing2"].write<std::vector<float>>();
+  std::vector<float>& fRing3 = tree["fRing3"].write<std::vector<float>>();
+  std::vector<float>& fRing4 = tree["fRing4"].write<std::vector<float>>();
+  std::vector<float>& fRing5 = tree["fRing5"].write<std::vector<float>>();
+  std::vector<float>& fRing6 = tree["fRing6"].write<std::vector<float>>();
+  std::vector<float>& fRing7 = tree["fRing7"].write<std::vector<float>>();
+  std::vector<float>& fRing8 = tree["fRing8"].write<std::vector<float>>();
+  std::vector<float>& nCh = tree["nCh"].write<std::vector<float>>();
+  std::vector<float>& nNeutrals = tree["nNeutrals"].write<std::vector<float>>();
+  std::vector<float>& ptD = tree["ptD"].write<std::vector<float>>();
+  std::vector<bool>& isMatched = tree["isMatched"].write<std::vector<bool>>();
```

Comment:
Part of what I liked about the ntuple solution (previous style) was that it removed the need for long lists of variables to be added directly into the code. Instead they were kept elsewhere and access from the "ntuple->" method. Is there any way to do that here? i.e. Can we move these to a new file or otherwise merge them into a new object? If not then this is fine; I think overall this is a huge improvement.